angular-c3.js

```diff
@@ -5,27 +5,139 @@ angular.module('c3', [])
 
 .directive('c3', function() {
   console.log('loading directive');
+
+  function getUnload(prev, curr) {
+    var prevKeys;
+    if (prev && prev.columns && prev.columns.length) {
+      prevKeys = {};
+      prev.columns.forEach(function(data) {
+        prevKeys[data[0]] = true;
+      });
+    } else if (prev && prev.rows && prev.rows.length) {
+      prevKeys = {};
+      prev.rows[0].forEach(function(key) {
+        prevKeys[key] = true;
+      });
+    } else {
+      return false;
+    }
+
+    var currKeys;
+    if (curr && curr.columns && curr.columns.length) {
+      currKeys = {};
+      curr.columns.forEach(function(data) {
+        currKeys[data[0]] = true;
+      });
+    } else if (curr && curr.rows && curr.rows.length) {
+      currKeys = {};
+      curr.rows[0].forEach(function(key) {
+        currKeys[key] = true;
+      });
+    } else {
+      return Object.keys(prevKeys);
+    }
+
+    var delta = [];
+    Object.keys(prevKeys).forEach(function(key) {
+      if (!currKeys.hasOwnProperty(key)) {
+        delta.push(key);
+      }
+    });
+
+    return (delta.length ? delta : false);
+  }
+
+  function doHide(chart, data) {
+    var ids;
+    if (data.columns) {
+      ids = data.columns.map(function(d) { return d[0]; });
+    } else if (data.rows) {
+      ids = data.rows[0];
+    }
```

Comment:
`json` and `url` are not handled. Will need to deal with that in a future release.